spring-pulsar/src/test/java/org/springframework/pulsar/core/reactive/DefaultReactiveMessageSenderFactoryTests.java

```diff
@@ -0,0 +1,159 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.core.reactive;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.pulsar.client.api.MessageId;
+import org.apache.pulsar.client.api.MessageRouter;
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.ProducerBuilder;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.api.TypedMessageBuilder;
+import org.apache.pulsar.reactive.client.api.MessageSpec;
+import org.apache.pulsar.reactive.client.api.MutableReactiveMessageSenderSpec;
+import org.apache.pulsar.reactive.client.api.ReactiveMessageSender;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * Common tests for {@link DefaultReactivePulsarSenderFactory}
+ *
+ * @author Christophe Bornet
+ */
+@SuppressWarnings("unchecked")
+class DefaultReactiveMessageSenderFactoryTests {
+
+	protected final Schema<String> schema = Schema.STRING;
+
+	private ProducerBuilder<String> producerBuilder;
+
+	private PulsarClient pulsarClient;
+
+	@BeforeEach
+	void createPulsarClient() {
+		pulsarClient = mock(PulsarClient.class);
+		producerBuilder = mock(ProducerBuilder.class);
+		Producer<String> producer = mock(Producer.class);
+		TypedMessageBuilder<String> mockMessage = mock(TypedMessageBuilder.class);
+
+		when(mockMessage.sendAsync()).thenReturn(CompletableFuture.completedFuture(MessageId.latest));
+		when(producer.newMessage()).thenReturn(mockMessage);
+		when(producer.closeAsync()).thenReturn(CompletableFuture.completedFuture(null));
+		when(producerBuilder.createAsync()).thenReturn(CompletableFuture.completedFuture(producer));
+		when(pulsarClient.newProducer(schema)).thenReturn(producerBuilder);
+	}
+
+	@Test
+	void createProducerWithSpecificTopic() {
+		ReactivePulsarSenderFactory<String> senderFactory = new DefaultReactivePulsarSenderFactory<>(pulsarClient,
+				null);
+		ReactiveMessageSender<String> sender = senderFactory.createReactiveMessageSender("topic1", schema);
+		sender.sendMessage(Mono.just(MessageSpec.of("test"))).block(Duration.ofSeconds(5));
```

Comment:
I would consider using `StepVerifier` instead of the block.

Reply:
Yes I was also considering it. Will take a look.

Reply:
Fair enough. Thx for looking into it.